pkg/azurefile/controllerserver.go

```diff
@@ -218,12 +218,16 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		return nil, status.Errorf(codes.InvalidArgument, fmt.Sprintf("matchTags must set as false when storageAccount(%s) is provided", account))
 	}
 
-	if subsID != "" && subsID != d.cloud.SubscriptionID {
+	if subsID == "" {
+		subsID = d.cloud.SubscriptionID
+	}
+
+	if subsID != d.cloud.SubscriptionID {
 		if fsType == nfs || protocol == nfs {
```

Comment:
I think we could remove this restriction also since  nfs should support cross subs.